code/modules/library/lib_items.dm

```diff
@@ -283,6 +284,68 @@
 	else
 		return ..()
 
+/obj/item/weapon/book/mindbook
+	name = "A Harmless Book"
+	icon = 'icons/obj/library.dmi'
+	icon_state ="book"
+	unique = 1
+	forbidden = 1
+	var/hallucinatory_cooldown
+
+/obj/item/weapon/book/mindbook/attack_self(mob/living/H)
+	if(ishuman(H))
+		var/mob/living/carbon/human/P = H
+		if((P.mind && P.mind.assigned_role == "Psychiatrist") || prob(50))

```

Comment:
If you get cloned you lose your mind's assigned role, which makes this not work.

We should probably fix that anyway, but you should probably know.